src/pandas_profiling/utils/common.py

```diff
@@ -88,3 +95,41 @@ def convert_timestamp_to_datetime(timestamp: int) -> datetime:
         return datetime.fromtimestamp(timestamp)
     else:
         return datetime(1970, 1, 1) + timedelta(seconds=int(timestamp))
+
+
+def test_for_pyspark_pyarrow_incompatibility():
+    """
+    This code checks if the env is using pyspark 2.3/2.4 with pyarrow >= 0.15, but has not set ARROW_PRE_0_15_IPC_FORMAT to 1
+    This will cause all toPandas() functions to break. If the env fits the pattern above, this code raises an warning
+    in warnings module in order to let the user know how to fix this problem
+
+    """
+
+    try:
+        # check versions
+        import os
+
+        import pyarrow
+        import pyspark
+
+        # get spark version as a tuple
+        spark_version = pyspark.__version__.split(".")
+
+        # if spark version == 2.3 or 2.4
+        if spark_version[0] == "2" and spark_version[1] in ["3", "4"]:
+
+            pyarrow_version = pyarrow.__version__.split(".")
+
+            # this checks if the env has an incompatible arrow version (not < 0.15)
+            if not (pyarrow_version[0] == "0" and int(pyarrow_version[1]) < 15):
+                # if os variable is not set, likely unhandled
+                if "ARROW_PRE_0_15_IPC_FORMAT" not in os.environ:
```

Comment:
Can be simplified as:
```
if "ARROW_PRE_0_15_IPC_FORMAT" not in os.environ or os.environ["ARROW_PRE_0_15_IPC_FORMAT"] != 1:
```

(or `os.environ.get("ARROW_PRE_0_15_IPC_FORMAT", 0) != 1`)

Reply:
thanks! this is way cleaner